hummingbot/connector/exchange/bloxroute_openbook/bloxroute_openbook_exchange.py

```diff
@@ -264,61 +277,73 @@ async def _place_order(
 
-        # this is temporarily hard coded to a single solana wallet
-        base_addr = CONSTANTS.TOKEN_PAIR_TO_WALLET_ADDR[base]
-        quote_addr = CONSTANTS.TOKEN_PAIR_TO_WALLET_ADDR[quote]
-        payer_address = base_addr if side == api.Side.S_ASK else quote_addr
+        payer_address = self._payer_address(base, quote, blxr_side)
+        open_orders_address = ""
+        if trading_pair in self._open_orders_address_mapper:
+            open_orders_address = self._open_orders_address_mapper[trading_pair]
 
         blxr_client_order_id = convert_hummingbot_to_blxr_client_order_id(order_id)
-        self._hummingbot_to_solana_id[order_id] = blxr_client_order_id
+        self._order_id_mapper[order_id] = blxr_client_order_id
 
-        submit_order_response = await self._provider_1.submit_order(
+        await self._testnet_provider.wait_connect()
+        post_order_response = await self._testnet_provider.post_order(
             owner_address=self._sol_wallet_public_key,
             payer_address=payer_address,
             market=trading_pair,
-            side=side,
-            types=[type],
+            side=blxr_side,
+            type=[blxr_order_type],
             amount=float(amount),
             price=float(price),
-            project=OPENBOOK_PROJECT,
-            client_order_id=blxr_client_order_id,
-            open_orders_address=self._open_orders_address,
+            open_orders_address=open_orders_address,
+            client_order_i_d=blxr_client_order_id,
+            project=constants.SPOT_OPENBOOK_PROJECT,
+        )
+
+        signed_tx = signing.sign_tx_with_private_key(post_order_response.transaction.content, self._key_pair)
+        post_submit_response = await self._testnet_provider.post_submit(
+            transaction=api.TransactionMessage(content=signed_tx),
             skip_pre_flight=True,
         )
 
-        self.logger().info(f"placed order {submit_order_response} with id {blxr_client_order_id}: {amount} @ {price}")
+        if open_orders_address == "":
+            self._open_orders_address_mapper[trading_pair] = post_order_response.open_orders_address
 
-        return submit_order_response, time.time()
+        return post_submit_response.signature, time.time()
+
+    def _payer_address(self, base, quote, blxr_side):
+        if blxr_side == api.Side.S_ASK:
+            return self._token_accounts[base]
+        return self._token_accounts[quote]
 
     async def _place_cancel(self, order_id: str, tracked_order: InFlightOrder):
-        if order_id not in self._hummingbot_to_solana_id:
-            raise Exception("placed order not found")
-        blxr_client_order_id = self._hummingbot_to_solana_id[order_id]
+        blxr_client_order_id = convert_hummingbot_to_blxr_client_order_id(order_id)
+        if tracked_order.trading_pair not in self._open_orders_address_mapper:
+            raise Exception("have to place an order before cancelling it") # TODO support this
```

Comment:
I think comment can be deleted? what would you expect to do if it doesn't exist?

Reply:
If there are existing orders, they are tried to be canceled before the strategy starts.

Reply:
But the problem with this is that when I tried to initialize the open orders addresses, it took too long to get the open orders and the exchange had already tried cancelling the orders before they were initialized.